src/prelude.rs

```diff
@@ -0,0 +1,755 @@
+    }
+
+    pub fn call(f: &str, xs: Vec<Expr>) -> Expr {
+        Expr::Call(span!(), f.into(), xs)
+    }
+}
+
+pub mod fact {
+    use super::*;
+
+    pub fn equals(x: Expr, y: Expr) -> Fact {
+        Fact::Eq(span!(), x, y)
+    }
+}
+
+pub mod sort {
+    use super::*;
+
+    pub fn i64() -> ArcSort {
+        I64Sort.to_arcsort()
+    }
+}
+
+/// Create a new ruleset.
+pub fn add_ruleset(egraph: &mut EGraph, ruleset: &str) -> Result<Vec<String>, Error> {
+    egraph.run_program(vec![Command::AddRuleset(span!(), ruleset.into())])
+}
+
+/// Run one iteration of a ruleset.
+pub fn run_ruleset(egraph: &mut EGraph, ruleset: &str) -> Result<Vec<String>, Error> {
+    egraph.run_program(vec![Command::RunSchedule(Schedule::Run(
+        span!(),
+        RunConfig {
+            ruleset: ruleset.into(),
+            until: None,
+        },
+    ))])
+}
+
+#[macro_export]
+macro_rules! vars {
+    [$($x:ident : $t:ident),* $(,)?] => {
+        &[$((stringify!($x), sort::$t())),*]
+    };
+}
+
+#[macro_export]
+macro_rules! expr {
+    ((unquote $unquoted:expr)) => { $unquoted };
+    (($func:tt $($arg:tt)*)) => { expr::call(stringify!($func), vec![$(expr!($arg)),*]) };
+    ($value:literal) => { expr::int($value) };
+    ($quoted:tt) => { expr::var(stringify!($quoted)) };
+}
+
+#[macro_export]
+macro_rules! fact {
+    ((= $($arg:tt)*)) => { fact::equals($(expr!($arg)),*) };
+    ($a:tt) => { Fact::Fact(expr!($a)) };
+}
+
+#[macro_export]
+macro_rules! facts {
+    ($($tree:tt)*) => { Facts(vec![$(fact!($tree)),*]) };
+}
+
+#[macro_export]
+macro_rules! action {
+    ((let $name:ident $value:tt)) => {
+        Action::Let(span!(), Symbol::from(stringify!($name)), expr!($value))
+    };
+    ((set ($f:ident $($x:tt)*) $value:tt)) => {
+        Action::Set(span!(), Symbol::from(stringify!($f)), vec![$(expr!($x)),*], expr!($value))
+    };
+    ((delete ($f:ident $($x:tt)*))) => {
+        Action::Change(span!(), Change::Delete, Symbol::from(stringify!($f)), vec![$(expr!($x)),*])
+    };
+    ((subsume ($f:ident $($x:tt)*))) => {
+        Action::Change(span!(), Change::Subsume, Symbol::from(stringify!($f)), vec![$(expr!($x)),*])
+    };
+    ((union $x:tt $y:tt)) => {
+        Action::Union(span!(), expr!($x), expr!($y))
+    };
+    ((panic $message:literal)) => {
+        Action::Panic(span!(), $message.to_owned())
+    };
+    ($x:tt) => {
+        Action::Expr(span!(), expr!($x))
+    };
+}
+
+#[macro_export]
+macro_rules! actions {
+    ($($tree:tt)*) => { GenericActions(vec![$(action!($tree)),*]) };
+}
+
+/// Add a rule to the e-graph whose right-hand side is made up of actions.
+pub fn rule(
+    egraph: &mut EGraph,
+    ruleset: &str,
+    facts: Facts<Symbol, Symbol>,
+    actions: Actions,
+) -> Result<Vec<String>, Error> {
+    let rule = Rule {
+        span: span!(),
+        head: actions,
+        body: facts.0,
+    };
+
+    let rule_name = egraph.parser.symbol_gen.fresh(&"prelude::rule".into());
```

Comment:
Would it be better if the rule name were a stringification of the actual rule?

We should also consider allowing the user to optionally pass a rule name if they want.